core/unit_test/TestMDRangeReduce.hpp

```diff
@@ -0,0 +1,61 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <gtest/gtest.h>
+
+#include <Kokkos_Core.hpp>
+
+namespace Test {
+
+template <typename T>
+void MDRangeReduceTester(int bound, int k) {
+  const auto policy_MD = Kokkos::MDRangePolicy<Kokkos::Rank<2>, TEST_EXECSPACE>(
+      {0, 0}, {bound, 2});
+
+  // No explicit fence() calls needed because result is in HostSpace
+  {
+    T lor_MD = 0;
+    Kokkos::parallel_reduce(
+        policy_MD,
+        KOKKOS_LAMBDA(const int i, const int, T& res) { res = res || i == k; },
```

Comment:
Might need to guard for LAMBDA support enabled.

Reply:
Earnest question: why is that still optional?

Reply:
I think technically if one configure with `-DKokkos_ENABLE_CUDA_LAMBDA=OFF` we do not pass the `-extended-lambda` flag to NVCC

Reply:
https://github.com/kokkos/kokkos/blob/fc4a9cecfa4d309881847626ebff718bbbe6af8c/cmake/kokkos_arch.cmake#L169-L174

Reply:
So, it's optional because Nvidia still considered it 'experimental'?

In CUDA 12, it's not experimental anymore:
https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html
Would we be averse to a PR that deprecates the CMake option, and unconditionally adds the version-appropriate flag and the `#define`?

Is there any benefit to users at this point of being able to *disable* it?

Reply:
You can bring it up in the developer meeting. 🙂

Reply:
#5964